### PR TITLE
Fix for "Project Contributor Count is broken" #420

### PR DESCRIPTION
**Pull request by Matthias-Schaub**
Firebase Functions
Refactor counter for first time user contribution to a project into seperate database trigger function.

Mapswipe Workers  Base Class:
Init contributorCount with 0.

### CODE REVIEW

File: firebase/functions/index.js [hunk 1]
```diff
@@ -163,6 +130,21 @@ exports.counter = functions.database.ref('/v2/results/{projectId}/{groupId}/{use
 })
 
 
+exports.counter_2 = functions.database.ref('/v2/users/{userId}/contributions/{projectId}/').onCreate((snapshot, context) => {
```
Comment (Hagellach37) [inline firebase/functions/index.js:133]: I like the approach to put this into a separate function.

Maybe you can find a better name than `counter_2`. :)  e.g. `projectCounter` or something similar.

**Comment by Matthias-Schaub** [comment]
I restructured (copy passed) code blocks in index.js. There are not that many real changes although git highlighted a lot.

**Review by Hagellach37 — APPROVED** [review]
Hey,
looks good to me. A general question: Do you think that there is further potential to untangle the first counter? it is doing a lot at the same time...

Maybe you can at least provide some comments for each of the counter function to describe the desired behaviour directly in this file. :)

**Comment by Matthias-Schaub** [comment]
I agree that the functions could be simpler and structured better to enhance readability. I can spent some more time on the functions to improve readability and documentation. With this pull request I just wanted to fix the bug as soon as possible.

File: firebase/functions/index.js (hunk 1 above)
Comment (Matthias-Schaub) [reply firebase/functions/index.js:133]: Yeah counter_2 is not an elegant or meaningful name :D
I will think of something better and as you said include some more comments and maybe restructure more code if that improves readability. It took me some time to figure out again which attribute / db ref is depended on which other and what will execute when.

**Comment by Hagellach37** [comment]
okay, agreed. Let's merge this one and address the other things separately. :)

**Pull request merged**